tests/misc/tstreamsreadline.nim

```diff
@@ -0,0 +1,27 @@
+discard """
+  output: ""
+"""
+
+# bug #9026
+import streams
+
+iterator readFile*(stream: Stream): int =
+  var num = 0
+  var line = ""
+  while stream.readLine(line):
+    num += 1
+    yield line.len
+
+iterator readFile*(s: string): int =
+  for x in readFile(newStringStream(s)):
+    yield x
+
+let
+  filenames = ["bin.txt", "str.txt"]
+  expected = [14334, 20773]
+
+for i, filename in filenames:
+  var lines = 0
+  for x in readFile(newFileStream(filename)):
+    lines += 1
+  doAssert lines == expected[i]
```

Comment:
also need to assert on content of 2 lines (eg 1st + last) to make sure line is exactly what we expect (makes sure it works for corner cases like line ending and end of file)